### ISSUE TITLE
Features file for Heterogeneous Graph Network

### ISSUE THREAD

**Issue by meenuagarwal** [williamleif/GraphSAGE#92]
I am trying to create input data files for the model but can't understand how to creat feat.npy file if I have two kinds of node in my graph (Let's say User and Item). Since I have 20 user attributes for every user and 50 item attributes for every item, how to create feat matrix for such graph? I would really appreciate your help on this.

**Comment by Jhy1993** [comment]
@meenuagarwal  Hi, do you find a solution? Maybe you can try N1*d1 for user and N1*d2 for item, then project them into a unified space(e.g., N1*d, N2*d).

**Comment by kuhanw** [comment]
Hi,

I am interested in this as well. My thought was similar to @Jhy1993. Concatenate the features of the node types together, maybe you can include a one hot encoding of the node type in addition as wel.

Cheers,

Kuhan

**Comment by RexYing** [comment]
Yes usually concatenating the node features with a one hot of whether the node is user or item would work.

**Comment by bjuthjliu** [comment]
Hi @meenuagarwal , I meet the same problem, have you solved the problem?

Thanks
